src/main/java/com/facebook/openwifirrm/ucentral/informationelement/NeighborReport.java

```diff
@@ -0,0 +1,344 @@
+import java.util.List;
+import java.util.Objects;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonElement;
+
+/**
+ * This information element (IE) appears in wifiscan entries. It currently does
+ * not appear in these entries AFAICT. It's called "Neighbor Report" in
+ * 802.11 specs (section 9.4.2.36). Refer to the specification for more details.
+ * Language in javadocs is taken from the specification.
+ */
+public class NeighborReport {
+	/** Defined in 802.11 table 9-92 */
+	public static final int TYPE = 52;
+
+	/**
+	 * The BSSID Information field can be used to help determine neighbor service
+	 * set transition candidates
+	 */
+	public static class BssidInfo {
+		/**
+		 * The capability subelement containing selected capability information for
+		 * the AP indicated by this BSSID.
+		 */
+		public static class Capabilities {
+			/** dot11SpectrumManagementRequired */
+			public final boolean spectrumManagement;
+			/** dot11QosOptionImplemented */
+			public final boolean qos;
+			/** dot11APSDOptionImplemented */
+			public final boolean apsd;
+			/** dot11RadioMeasurementActivated */
+			public final boolean radioMeasurement;
+
+			/** Constructor */
+			public Capabilities(
+				boolean spectrumManagement,
+				boolean qos,
+				boolean apsd,
+				boolean radioMeasurement
+			) {
+				this.spectrumManagement = spectrumManagement;
+				this.qos = qos;
+				this.apsd = apsd;
+				this.radioMeasurement = radioMeasurement;
+			}
+
+			/** Parse Capabilities from JSON object */
+			// TODO rename fields as necessary - we don't know how the data format yet
+			public static Capabilities parse(JsonObject contents) {
+				return new Capabilities(
+					contents.get("Spectrum Management").getAsBoolean(),
+					contents.get("QoS").getAsBoolean(),
+					contents.get("APSD").getAsBoolean(),
+					contents.get("Radio Management").getAsBoolean()
+				);
+			}
+
+			@Override
+			public int hashCode() {
+				return Objects
+					.hash(spectrumManagement, qos, apsd, radioMeasurement);
+			}
+
+			@Override
+			public boolean equals(Object obj) {
+				if (obj == null) {
+					return false;
+				}
+
+				if (this == obj) {
+					return true;
+				}
+
+				if (getClass() != obj.getClass()) {
+					return false;
+				}
+
+				Capabilities other = (Capabilities) obj;
+				return spectrumManagement == other.spectrumManagement &&
+					qos == other.qos && apsd == other.apsd &&
+					radioMeasurement == other.radioMeasurement;
+			}
+
+			@Override
+			public String toString() {
+				return String.format(
+					"Capabilities[spectrumManagement=%d, qos=%d, apsd=%d, radioMeasurement=%d]",
+					spectrumManagement,
+					qos,
+					apsd,
+					radioMeasurement
+				);
+			}
+		}
+
+		/**
+		 * 2 unsigned bits - whether the AP identified by this BSSID is reachable by
+		 * the STA that requested the neighbor report
+		 */
+		public final byte apReachability;
+		/**
+		 * if 1, indicates that the AP identified by this BSSID supports the same
```

Comment:
Capitalize javadoc and all others (in this and other classes, in case I missed any)